scripts/install/jdownloader.sh

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
+# JDownloader Installer for swizzin
+# Author: Aethaeran
+
+# References
+# https://swizzin.ltd/dev/structure/
+# https://www.digitalocean.com/community/tutorials/how-to-install-java-with-apt-on-debian-10
+# https://support.jdownloader.org/Knowledgebase/Article/View/install-jdownloader-on-nas-and-embedded-devices
+# https://support.jdownloader.org/Knowledgebase/Article/View/headless-systemd-autostart-script
+# Liara already made a doc for installing JDownloader manually https://docs.swizzin.net/guides/jdownloader/
+# https://linuxize.com/post/how-to-check-if-string-contains-substring-in-bash/
+# https://linuxize.com/post/bash-check-if-file-exists/
+# https://superuser.com/questions/402979/kill-program-after-it-outputs-a-given-line-from-a-shell-script
+
+function get_myjd_info() {
+
+    # TODO: Give the option to skip this, and have user do it manually later.
+
+    echo_info "An account from https://my.jdownloader.org/ is required in order to access the web UI.\nUse a randomly generated password at registration as the password is stored in plain text"
+    if [[ -z "${MYJD_EMAIL}" ]]; then
+        echo_query "Please enter the e-mail used to access this account once created:"
+        read -r 'MYJD_EMAIL'
+    else
+        echo_info "Using email = $MYJD_EMAIL"
+    fi
+
+    if [[ -z "${MYJD_PASSWORD}" ]]; then
+        echo_query "Please enter the password for the account"
+        read -r 'MYJD_PASSWORD'
+    else
+        echo_info "Using password = $MYJD_PASSWORD"
+    fi
+
+    if [[ -z "${MYJD_DEVICENAME}" ]]; then
+        echo_query "Please enter the desired device name"
+        read -r 'MYJD_DEVICENAME'
+    else
+        echo_info "Using device name = $MYJD_DEVICENAME"
+    fi
+
+    mkdir -p "$JD_HOME/cfg"
+
+    if [[ -e "$JD_HOME/cfg/org.jdownloader.api.myjdownloader.MyJDownloaderSettings.json" ]]; then
+        rm "$JD_HOME/cfg/org.jdownloader.api.myjdownloader.MyJDownloaderSettings.json"
+    fi
+
+    cat > "$JD_HOME/cfg/org.jdownloader.api.myjdownloader.MyJDownloaderSettings.json" << EOF
+{
+    "email" : "$MYJD_EMAIL",
+    "password" : "$MYJD_PASSWORD",
+    "devicename" : "$MYJD_DEVICENAME"
+}
+EOF
+}
```

Comment:
this *could* be something in a functions file.

Reply:
Will likely be. Going to add a catch to inject MyJDownloader details into installations that don't have proper details yet.

Might have to move some of the MyJD verification into this function as well.